src/Util/Sniffs/TokenIs.php

```diff
@@ -0,0 +1,296 @@
+<?php
+/**
+ * Utility functions to determine what an ambiguous token represents.
+ *
+ * @author    Greg Sherwood <gsherwood@squiz.net>
+ * @author    Juliette Reinders Folmer <phpcs_nospam@adviesenzo.nl>
+ * @copyright 2006-2019 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Util\Sniffs;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+class TokenIs
```

Comment:
I understand your objection and I agree it's a bit of an odd one out. It is basically a class to deal with ambiguous tokens.

Would renaming it to `AmbiguousToken` help ?

Reply:
> Would renaming it to `AmbiguousTokens` help ?

I don't think that's the right name.

My original comment about this was:
> I originally thought that maybe all the "tokenIs" style methods should be grouped together. So you end up with Token::isNameSpaceDeclaration() and Token::isShortList() and Token:: isOOConstant, but you then lose the grouping of the "tokenIs" methods with their other related methods. I'm still trying to figure out if that is worse than ending up with methods that have no proper class to live in.

I'm not sure that moving all the methods around is the right thing to do, but it would make all the `Token::isSomething()` calls easier to read. Maybe the `TokenIs` class (or a new name) is a wrapper for all those `isSomething()` methods, and just calls out to the other util classes. But when there isn't an obvious util class to call out to, it defines the method itself.

Reply:
I would agree that a sniff writer's usage should all be `Token::isSomething()`, regardless of what that Something is. I shouldn't need to look up whether it is `Token::` or `TokenIs::`.

Internally, that can contain the logic directly, or defer to methods in this class - maybe `TokenClarifier::`, or `TokenInstance::`?

Reply:
How about renaming the method names? Or is that too much of a breaking change. For instance instead of `TokenIs::isShortList()` have `TokenIs::shortList()` ?

Reply:
I got with `-er` in this case, as it's a service at first sight, not a value object/data object/entity.

- `TokenAnalyzer` (is-*, read-only)
- `TokenResolver` (from A to B)

From suggestions above I like one from @GaryJones 

- `TokenClarifier`

Reply:
I commented above that maybe `TokenUtil` is a good name for this class, although that was in the context of the `StringUtil` discussion. I could see people writing code like `TokenUtil::isTokenShortList()`.